### PR TITLE
Chore/Write docstrings for python classes

### PR DESCRIPTION
**Pull request by jamesL92**
First pass of improving the `pdoc` documentation.

### CODE REVIEW

**Comment by jamesL92** [comment]
Should help improve the situation of #127

**Comment by jamesL92** [comment]
Some docstrings have been taken from https://configu.com/docs/, others I've had a go at writing based on my understanding on various concepts here.

Wouldn't be surprised if I'm misunderstanding any of the concepts here, if so please let me know and I can amend the docstrings accordingly.

**Comment by jamesL92** [comment]
Hi @RonConfigu 
I've rebased this on top of the main branch, should be ready for a review now :+1:

File: py/configu/core/config_schema.py [hunk 1]
```diff
@@ -86,6 +92,13 @@ def __init__(self, path: str) -> None:
         super().__init__(path=path, type=ConfigSchema.TYPES[Path(path).suffix])
 
     def read(self) -> str:
+        """
+        Reads the config schema file
+        :return: contents of the config schema
+        """
+        import pdb
+
+        pdb.set_trace()
```
Comment (rannn505) [inline py/configu/core/config_schema.py:101]: @jamesL92 is it a here by mistake?

File: py/configu/commands/eval_command.py [hunk 2]
```diff
@@ -68,11 +68,25 @@ class EvaluatedConfigOrigin(Enum):
...
+        """
+        Creates a new EvalCommand.
+        :param parameters: dict
+            The command's parameters. Includes the following:
+            - from: a dataclass containing the following:
+                - store: the `ConfigStore` from which to fetch
+                - set: the `ConfigSet` to fetch
+                - schema: `ConfigSchema` to validate the config being fetched
+            - configs (optional): a dictionary of overrides to the fetched
+            `Config`s
```
Comment (rannn505) [inline py/configu/commands/eval_command.py:88]:
```suggestion
            The command's parameters. Includes the following:
             - store: the `ConfigStore` from which to fetch
             - set: the `ConfigSet` to fetch
             - schema: `ConfigSchema` to validate the config being fetched
             - configs (optional): a dictionary of overrides to the fetched`Config`s
```

Comment (rannn505) [reply py/configu/commands/eval_command.py:88]: The API changed a bit since this PR :)

Comment (jamesL92) [reply py/configu/commands/eval_command.py]: And so it has - Docstrings updated :+1:

File: py/configu/core/config_schema.py (hunk 1 above)
Comment (jamesL92) [reply py/configu/core/config_schema.py]: it is indeed :man_facepalming: - removed.

**Comment by jamesL92** [comment]
Changes made - would be happy if someone with write access can merge this change in :)

**Comment by RonConfigu** [comment]
🥇

**Pull request merged**